admin_guide/sysctls.adoc

```diff
@@ -0,0 +1,168 @@
+[[admin-guide-sysctls]]
+= Sysctls
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+Certain sysctl settings are exposed via Kubernetes, allowing users to modify
+kernel parameters at runtime for namespaces within a container. Only sysctls
+that are namespaced can be set independently on pods; if a sysctl is not
+namespaced (called _node-level_), it cannot be set within {product-title}.
+
+ifdef::openshift-enterprise[]
+[NOTE]
+====
+As of {product-title} 3.3.1, sysctl support is a feature in
+link:https://access.redhat.com/support/offerings/techpreview[Technology
+Preview].
+====
+endif::[]
+
+[[undersatnding-sysctls]]
+== Understanding Sysctls
+
+In Linux, the sysctl interface allows an administrator to modify kernel
+parameters at runtime. Parameters are available via the *_/proc/sys/_* virtual
+process file system. The parameters cover various subsystems such as:
+
+- kernel (common prefix: *_kernel._*)
+- networking (common prefix: *_net._*)
+- virtual memory (common prefix: *_vm._*)
+- MDADM (common prefix: *_dev._*)
+
+More subsystems are described in
+link:https://www.kernel.org/doc/Documentation/sysctl/README[Kernel documentation]. To get a list of all parameters, you can run:
+
+----
+$ sudo sysctl -a
+----
+
+[[namespaces-vs-node-level-sysctls]]
+== Namespaced vs Node-Level Sysctls
+
+A number of sysctls are _namespaced_ in today’s Linux kernels. This means that
+they can be set independently for each pod on a node. Being namespaced is a
+requirement for sysctls to be accessible in a pod context within Kubernetes.
+
+The following sysctls are known to be namespaced:
+
+- *_kernel.shm*_*
+- *_kernel.msg*_*
+- *_kernel.sem_*
+- *_fs.mqueue.*_*
+- *_net.*_*
+
+Sysctls which are not namespaced are called _node-level_ and must be set
+manually by the cluster administrator, either by means of the underlying Linux
+distribution of the nodes (e.g., via *_/etc/sysctls.conf_*) or using a DaemonSet
+with privileged containers.
+
+[NOTE]
+====
+It is good practice to consider nodes with special sysctl settings as tainted
+within a cluster, and only schedule pods onto them which need those sysctl
+settings. It is suggested to use the
+link:http://kubernetes.io/docs/user-guide/kubectl/kubectl_taint/[Kubernetes
+_taints and toleration_ feature] to implement this.
+====
```

Comment:
Expletive sentence x2. Maybe:

Consider nodes with special sysctl settings as tainted. Only schedule pods onto them that need those sysctl settings. Use the Kubernetes taints and toleration feature to implement this.

?